Adds Contentful web provider

jerriep opened the pull request:
Adds support for Contentful:

![image](https://github.com/user-attachments/assets/93b5b40f-9d7c-4094-b6b7-7d142898bed3)

jerriep committed:
Adds Contentful web provider

jerriep commented:
@kevinchalet It is not clear to me from the build output why the tests are failing. All the tests on my local computer is passing

kevinchalet commented on src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.Userinfo.cs:
```diff
@@ -519,6 +519,21 @@ public ValueTask HandleAsync(ExtractUserInfoResponseContext context)
                         context.Response[Claims.GivenName] = name["first"];
                     }
                 }
+                
```
In general, we only use this handler and fix up non-standard claims here for providers that implement a "standard OIDC userinfo" (or claim to 😄) but that have compliance issues (e.g the node type of a standard claim isn't correct). Otherwise, it's simpler/better not to amend the returned claims here and only do the non-standard -> WS-Federation claims mapping in the `MapCustomWebServicesFederationClaims` handler.

kevinchalet commented on src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationProviders.xml:
```diff
@@ -431,6 +431,23 @@
+                                ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
+  -->
+
+  <Provider Name="Contentful" DisplayName="Contentful" Id="3d820b1e-33c7-4934-a783-48745343d53b"
```
```suggestion
  <Provider Name="Contentful" Id="3d820b1e-33c7-4934-a783-48745343d53b"
```

Tip: when the display name matches the provider name, no need to set it explicitly 😃

kevinchalet commented:
Thanks for your PR!

> @kevinchalet It is not clear to me from the build output why the tests are failing. All the tests on my local computer is passing

Looks like a test was failing due to some crypto issues completely unrelated to your change:

![image](https://github.com/user-attachments/assets/7972effb-b52d-416c-9e5a-986cbb7eb0c1)

I restarted the build and it's green now 🤣

jerriep committed:
Responds to code review feedback

- Removes DisplayName attribute
- Moves claim resolution to MapCustomWebServicesFederationClaims

jerriep commented:
@kevinchalet I addressed the code review issues as requested

kevinchalet commented on src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.Userinfo.cs:
```diff
@@ -519,7 +519,7 @@ public ValueTask HandleAsync(ExtractUserInfoResponseContext context)
                     }
                 }
-
+                
```
```suggestion
```

kevinchalet committed:
Update src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.Userinfo.cs

kevinchalet commented on src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.Userinfo.cs:
```diff
@@ -519,7 +519,6 @@ public ValueTask HandleAsync(ExtractUserInfoResponseContext context)
                     }
                 }
-
                 return default;
```
```suggestion

                return default;
```

kevinchalet committed:
Update src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.Userinfo.cs

kevinchalet commented:
Merged. Thanks @jerriep!
